mmeval/metrics/instance_seg.py

```diff
@@ -0,0 +1,288 @@
+
+            groundtruths (Sequence[Dict]): A sequence of dict. Each dict
+                represents a groundtruths for an image, with the following
+                keys:
+
+                - pts_instance_mask(np.ndarray): Ground truth instance masks.
+                - pts_semantic_mask(np.ndarray): Ground truth semantic masks.
+        """
+        for prediction, groundtruth in zip(predictions, groundtruths):
+            self._results.append((deepcopy(prediction), deepcopy(groundtruth)))
+
+    def compute_metric(self, results: List[List[Dict]]) -> Dict[str, float]:
+        """Compute the metrics from processed results.
+
+        Args:
+            results (list): The processed results of each batch.
+
+        Returns:
+            Dict[str, float]: The computed metrics. The keys are the names of
+            the metrics, and the values are corresponding results.
+        """
+        gt_semantic_masks = []
+        gt_instance_masks = []
+        pred_instance_masks = []
+        pred_instance_labels = []
+        pred_instance_scores = []
+
+        for result_pred, result_gt in results:
+            gt_semantic_masks.append(result_gt['pts_semantic_mask'])
+            gt_instance_masks.append(result_gt['pts_instance_mask'])
+            pred_instance_masks.append(result_pred['pts_instance_mask'])
+            pred_instance_labels.append(result_pred['instance_labels'])
+            pred_instance_scores.append(result_pred['instance_scores'])
+
+        ret_dict = self.instance_seg_eval(
+            gt_semantic_masks,
+            gt_instance_masks,
+            pred_instance_masks,
+            pred_instance_labels,
+            pred_instance_scores,
+            valid_class_ids=self.valid_class_ids,
+            class_labels=self.classes)
+
+        return ret_dict
+
+    def aggregate_predictions(self, masks, labels, scores, valid_class_ids):
+        """Maps predictions to ScanNet evaluator format.
+
+        Args:
+            masks (list[np.ndarray]): Per scene predicted instance masks.
+            labels (list[np.ndarray]): Per scene predicted instance labels.
+            scores (list[np.ndarray]): Per scene predicted instance scores.
+            valid_class_ids (tuple[int]): Ids of valid categories.
+
+        Returns:
+            list[dict]: Per scene aggregated predictions.
+        """
+        infos = []
+        for id, (mask, label, score) in enumerate(zip(masks, labels, scores)):
+            info = dict()
+            n_instances = mask.max() + 1
+            for i in range(n_instances):
+                file_name = f'{id}_{i}'
+                info[file_name] = dict()
+                info[file_name]['mask'] = (mask == i).astype(int)
+                info[file_name]['label_id'] = valid_class_ids[label[i]]
+                info[file_name]['conf'] = score[i]
+            infos.append(info)
+        return infos
+
+    def rename_gt(self, gt_semantic_masks, gt_instance_masks, valid_class_ids):
```

Comment:
The `rename` in function name `rename_gt` seems a bit weird, what is be renamed?

Reply:
It seems that this function rename the input ground truth in order to evaluate in ScanNet protocol provided by [ScanNet repo](https://github.com/ScanNet/ScanNet).